### PR TITLE
Slayer plugin: update infobox counter whenever bracelet triggers

### PR DESCRIPTION
**Pull request by iguaan**
Currently infobox only updates whenever the killedOne() is triggered, this fixes that.

### CODE REVIEW

**Comment by iguaan** [comment]
No clue how travis failed with such a minor change.

**Comment by Reasel** [comment]
Wouldn't this add the counter even if you specifically set the infobox to not show up?

Also why would this ever be needed? Every kill triggers an update. By adding this you are updating twice. There is only one way to lower the current task, which is triggered by killedOn. I am not seeing the reason for this at all.

**Comment by iguaan** [comment]
> Wouldn't this add the counter even if you specifically set the infobox to not show up?

Good point, added necessary checks.


> Also why would this ever be needed?

Bracelet charge counts after killedOne() is triggered, this would make it so that actual task counter and the infobox counter would be off by one until another kill is done.

**Comment by Reasel** [comment]
Okay I see what you mean. In that case I would just bundle the code for checking and updating the infobox into a single function and call that instead as well as call it in killedOne(). Make sure to set the timer as well.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 1]
```diff
@@ -503,6 +496,17 @@ private void removeCounter()
...
 
+	private void updateCounter()
+	{
+		if (!config.showInfobox())
+		{
+			return;
+		}
+
+		addCounter();
+		counter.setText(String.valueOf(amount));
```
Comment (Reasel) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: Make sure to add `infoTimer = Instant.now();`

Comment (iguaan) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: I'm pretty sure I added it when I made the changes first time, but I had to re-do them. Thanks for pointing that out.

**Comment by Nightfirecat** [comment]
Needs rebase

**Comment by daviscook477** [comment]
ok so I killed fire giants until I got the slaughter to trigger and the tooltip indicated 131 after the slaughter triggered
tooltip_after_slaughter
in order to trigger an update I did an additional kill to get killedone to trigger
one_kill_after_slaughter_trigger
now it says 130 so it looks like it worked fine without needing any changes

**Comment by daviscook477** [comment]
triggering bracelet of slaughter
current_master_slaughter
after checking the gem to verify looks like the slaughter bracelet had no issue
current_master_check

so this isn't a bugfix. but I am going to recommend that we accept this pull request since it will help prevent bugs in the future. if in the future it ever becomes the case that the order of chat message event before killedone is broken this will become an actual bug unless this change is merged.

**Comment by Adam-** [comment]
This issue is no longer a problem on master, not sure when it got fixed.

**Pull request closed without merging**